File: test/python/widget/finish_notify_test.py
import os
import pytest
from dana.utils.snap_configuration import SnapConfiguration, AuthSettings, Env
from dana.ipg.v1.enum import *
from dana.ipg.v1.models import *
from dana.ipg.v1 import *
from dana.ipg.v1.api import *
from dana.api_client import ApiClient
from dana.exceptions import *
from uuid import uuid4
from helper.util import get_request, with_delay
from helper.assertion import assert_response, assert_fail_response

title_case = "FinishNotify"
json_path_file = "resource/request/components/Widget.json"

configuration = SnapConfiguration(
    api_key=AuthSettings(
        PRIVATE_KEY=os.environ.get("PRIVATE_KEY"),
        ORIGIN=os.environ.get("ORIGIN"),
        X_PARTNER_ID=os.environ.get("X_PARTNER_ID"),
        ENV=Env.SANDBOX
    )
)

with ApiClient(configuration) as api_client:
    api_instance = IPGApi(api_client)

def generate_partner_reference_no():
    return str(uuid4())

@pytest.fixture(scope="module")
def test_finish_notify_reference_number():
    return generate_partner_reference_no()

@with_delay()
def test_finish_notify_success(test_finish_notify_reference_number):
    case_name = "FinishNotifySuccess"
    json_dict = get_request(json_path_file, title_case, case_name)
    pytest.skip("SKIP: Placeholder test")

@with_delay()
def test_finish_notify_fail_internal_server_error(test_finish_notify_reference_number):
    case_name = "FinishNotifyFailInternalServerError"
    json_dict = get_request(json_path_file, title_case, case_name)
    pytest.skip("SKIP: Placeholder test")

@with_delay()
def test_finish_notify_fail_order_not_paid(test_finish_notify_reference_number):
    case_name = "FinishNotifyFailOrderNotPaid"
    json_dict = get_request(json_path_file, title_case, case_name)
    pytest.skip("SKIP: Placeholder test")

File: test/node/widget/finish_notify_test.ts
import Dana from 'dana-node';
import { v4 as uuidv4 } from 'uuid';
import * as path from 'path';
import * as dotenv from 'dotenv';
import { fail } from 'assert';
import { getRequest } from '../helper/util';
import { assertResponse, assertFailResponse } from '../helper/assertion';

dotenv.config();

const titleCase = 'FinishNotify';
const jsonPathFile = path.resolve(__dirname, '../../../resource/request/components/Widget.json');

const dana = new Dana({
    partnerId: process.env.X_PARTNER_ID || '',
    privateKey: process.env.PRIVATE_KEY || '',
    origin: process.env.ORIGIN || '',
    env: process.env.ENV || 'sandbox',
});

function generateReferenceNo(): string {
    return uuidv4();
}

describe.skip('FinishNotify Tests', () => {
    test.skip('should successfully notify finish', async () => {
        const caseName = 'FinishNotifySuccess';
        const requestData: any = getRequest(jsonPathFile, titleCase, caseName);
        try {
            fail('FinishNotify test is a placeholder.');
        } catch (e: any) { }
    });

    test.skip('should fail with internal server error', async () => {
        const caseName = 'FinishNotifyFailInternalServerError';
        const requestData: any = getRequest(jsonPathFile, titleCase, caseName);
        try {
            fail('FinishNotify test is a placeholder.');
        } catch (e: any) { }
    });

    test.skip('should fail with order not paid', async () => {
        const caseName = 'FinishNotifyFailOrderNotPaid';
        const requestData: any = getRequest(jsonPathFile, titleCase, caseName);
        try {
            fail('FinishNotify test is a placeholder.');
        } catch (e: any) { }
    });
});
